subsys/bluetooth/host/gatt.c

```diff
@@ -3523,16 +3522,23 @@ bool bt_gatt_is_subscribed(struct bt_conn *conn,
 		return false;
 	}
 
-	ccc = attr->user_data;
+	if (!attr->read) {
+		LOG_ERR("Read method not set");
+		return false;
+	}
 
-	/* Check if the connection is subscribed */
-	for (size_t i = 0; i < BT_GATT_CCC_MAX; i++) {
-		const struct bt_gatt_ccc_cfg *cfg = &ccc->cfg[i];
+	len = attr->read(conn, attr, &ccc_bits, sizeof(ccc_bits), 0);
```

Comment:
To treat it as an integer, we must parse the read result (as a little-endian integer). Something like the following:

```suggestion
	len = attr->read(conn, attr, ccc_bits_encoded, sizeof(ccc_bits), 0);
	ccc_bits = sys_get_le16(ccc_bits_encoded);
	
	// Declaration should be above. I put it here for a nicer diff.
	uint8_t ccc_bits_encoded[sizeof(ccc_bits)];
```

Feel free to experiment with helper functions here, if you want.

Reply:
Added, tested before and after using the ccc_store bsim test: https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/bluetooth/host/gatt/ccc_store/src/peripheral.c#L147, both works but your suggestion is more correct I think :)